packages/starknet-snap/src/state/account-state-manager.ts

```diff
@@ -172,6 +184,16 @@ export class AccountStateManager extends StateManager<AccContract> {
           throw new Error(`Account already exists`);
         }
 
+        if (
+          account.accountName &&
+          (await this.isAccountNameExist(
+            { accountName: account.accountName, chainId: account.chainId },
+            state,
+          ))
+        ) {
+          throw new Error(`Account name already exists`);
+        }
+
```

Comment:
not sure if we trim b4 or not, may be better to add a safe guard?

```suggestion
        const trimedAccountName = accountName?.trim();
        if (
          trimedAccountName &&
          (await this.isAccountNameExist(
            { accountName: trimedAccountName, chainId },
            state,
          ))
        ) {
          throw new Error(`Account name already exists`);
        }

```